errors/errors.go

```diff
@@ -35,28 +35,24 @@ var (
 	// ever be reached if the code was written as expected by the framework
 	ErrHuman = Register(7, "coding error")
 
-	// ErrCannotBeModified is returned when something that is considered immutable
+	// ErrImmutable is returned when something that is considered immutable
 	// gets modified
-	ErrCannotBeModified = Register(8, "cannot be modified")
+	ErrImmutable = Register(8, "cannot be modified")
 
 	// ErrEmpty is returned when a value fails a not empty assertion
 	ErrEmpty = Register(9, "value is empty")
 
-	// ErrInvalidState is returned when an object is in invalid state
-	ErrInvalidState = Register(10, "invalid state")
+	// ErrState is returned when an object is in invalid state
+	ErrState = Register(10, "invalid state")
 
-	// ErrInvalidType is returned whenever the type is not what was expected
-	ErrInvalidType = Register(11, "invalid type")
+	// ErrType is returned whenever the type is not what was expected
+	ErrType = Register(11, "invalid type")
 
-	// ErrInsufficientAmount is returned when an amount of currency is
-	// insufficient, e.g. funds/fees
-	ErrInsufficientAmount = Register(12, "insufficient amount")
+	// ErrAmount is returned when processed amount is invalid.
+	ErrAmount = Register(13, "invalid amount")
 
-	// ErrInvalidAmount stands for invalid amount of whatever
```

Comment:
So this one collapse to the above, right? Looks good

Reply:
Yes, there is only one now as I think generalizing it is good enough.